src/sys/sendfile.rs

```diff
@@ -68,24 +68,24 @@ cfg_if! {
                  target_os = "freebsd",
                  target_os = "ios",
                  target_os = "macos"))] {
-        use crate::sys::uio::IoVec;
+        use std::io::IoSlice;
 
         #[derive(Clone, Debug, Eq, Hash, PartialEq)]
```

Comment:
This is causing failures, because IoSlice doesn't implement some of these traits.  But `SendfileHeaderTrailer` isn't `pub`, so there's no penalty to remove some of its trait impls.  What happens if you remove the offending trait impls?  If it breaks something else, then you'll probably have to manually implement them, rather than derive them.